pkg/build/controller/controller.go

```diff
@@ -302,55 +302,56 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	nextStatus := build.Status.Phase
 	currentReason := build.Status.Reason
 
-	switch pod.Status.Phase {
-	case kapi.PodRunning:
-		// The pod's still running
-		build.Status.Reason = ""
-		build.Status.Message = ""
-		nextStatus = buildapi.BuildPhaseRunning
-
-	case kapi.PodPending:
-		build.Status.Reason = ""
-		build.Status.Message = ""
-		nextStatus = buildapi.BuildPhasePending
-		if secret := build.Spec.Output.PushSecret; secret != nil && currentReason != buildapi.StatusReasonMissingPushSecret {
-			if _, err := bc.SecretClient.Secrets(build.Namespace).Get(secret.Name); err != nil && errors.IsNotFound(err) {
-				build.Status.Reason = buildapi.StatusReasonMissingPushSecret
-				build.Status.Message = buildapi.StatusMessageMissingPushSecret
-				glog.V(4).Infof("Setting reason for pending build to %q due to missing secret %s/%s", build.Status.Reason, build.Namespace, secret.Name)
+	if build.Status.Phase != buildapi.BuildPhaseFailed {
```

Comment:
I tested that and the build object always matched the cache correctly, so I didn't think it was worth introducing that much new code for no reason.